packages/hash/frontend/src/blocks/page/Comments/CommentBlock.tsx

```diff
@@ -0,0 +1,186 @@
+import { FunctionComponent, useMemo, useState, useRef, useEffect } from "react";
+import { Box, Typography } from "@mui/material";
+import {
+  Avatar,
+  Button,
+  FontAwesomeIcon,
+  IconButton,
+} from "@hashintel/hash-design-system/ui";
+import { formatDistanceToNowStrict } from "date-fns";
+import {
+  faChevronDown,
+  faChevronUp,
+  faEllipsisVertical,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { usePopupState } from "material-ui-popup-state/hooks";
+import { bindTrigger } from "material-ui-popup-state";
+import { types } from "@hashintel/hash-shared/types";
+import { extractBaseUri } from "@blockprotocol/type-system-web";
+import { PageComment } from "../../../components/hooks/usePageComments";
+import { CommentTextField, CommentTextFieldRef } from "./CommentTextField";
+import { CommentBlockMenu } from "./CommentBlockMenu";
+import styles from "./style.module.css";
+
+const LINE_HEIGHT = 21;
+
+type ShowMoreTextLinkProps = {
+  label: string;
+  icon: IconDefinition;
+  onClick: () => void;
+};
+
+export const ToggleTextExpandedButton: FunctionComponent<
+  ShowMoreTextLinkProps
+> = ({ label, icon, onClick }) => (
+  <Button
+    size="xs"
+    variant="tertiary_quiet"
+    sx={{
+      display: "flex",
+      fontWeight: 600,
+      px: 0.5,
+      py: 0,
+      minHeight: 0,
+      color: ({ palette }) => palette.primary.main,
+      alignSelf: "flex-end",
+    }}
+    onClick={onClick}
+  >
+    {label}
+    <FontAwesomeIcon icon={icon} sx={{ fontSize: 12, ml: 0.75 }} />
+  </Button>
+);
+
+type CommentProps = {
+  comment: PageComment;
+};
+
+export const CommentBlock: FunctionComponent<CommentProps> = ({ comment }) => {
+  const { hasText, author, textUpdatedAt } = comment;
+
+  const contentRef = useRef<CommentTextFieldRef>(null);
+  const [collapsed, setCollapsed] = useState(true);
+  const [shouldCollapse, setShouldCollapse] = useState(false);
+
+  const commentCreatedAt = useMemo(() => {
+    const updatedAt = new Date(textUpdatedAt);
+    const timeDistance = formatDistanceToNowStrict(updatedAt);
+    return timeDistance === "0 seconds"
+      ? "Just now"
+      : `${formatDistanceToNowStrict(updatedAt)} ago`;
+  }, [textUpdatedAt]);
+
+  const commentMenuPopupState = usePopupState({
+    variant: "popover",
+    popupId: "comment-block-menu",
+  });
+
+  useEffect(() => {
+    if (contentRef.current) {
+      const dom = contentRef.current.getDom();
+      if (dom) {
+        setShouldCollapse(dom.scrollHeight > LINE_HEIGHT * 2);
+      }
+    }
+  }, []);
```

Comment:
You could have `CommentTextField` take a function which is called with the number of lines it takes up to handle this instead? Something like `onLineCountChange` or something? Much better than this kind of imperative code.

Reply:
Followed your suggestion and added `onLineCountChange` to `CommentTextField` in d58bfd2086d09f732e1b933d3a317c75af5ebe31. Do you agree with it being called inside `useLayoutEffect`?